protocols/gossipsub/src/behaviour.rs

```diff
@@ -1623,15 +1628,22 @@ where
         }
 
         for p in px {
```

Comment:
I can't add a comment higher up in the code, but you probably want to remove lines 1618 to 1622
```rust
// Ignore peerInfo with no ID
//
//TODO: Once signed records are spec'd: Can we use peerInfo without any IDs if they have a
// signed peer record?
px.retain(|p| p.peer_id.is_some());
```